FinancialTracker/Firebase.swift

```diff
@@ -56,7 +56,7 @@ class FirebaseHandler {
     }
     
     private init() {
-        if self.user == nil && signedIn {
+        if signedIn {
```

Comment:
It seems that you do not need `if signedIn` at all here. 

`getCurrentUserData` does this check `guard let uid = self.auth.currentUser?.uid` which in way us the same was what `signedIn` does. 

So, on of the two is not needed.